analyses/long-format-table-utils/annotator/download-annotation-data.R

```diff
@@ -0,0 +1,205 @@
+    stop(err_cond)
+  }
+)
+
+input_data_dir <- file.path(root_dir, "data")
+
+output_data_dir <- file.path(
+  root_dir, "analyses", "long-format-table-utils", "annotator",
+  "annotation-data")
+
+if (!dir.exists(output_data_dir)) {
+  dir.create(output_data_dir)
+}
+
+
+
+# Read input data --------------------------------------------------------------
+# ensg hugo rmtl mappings
+ensg_hugo_rmtl_df <- dplyr::distinct(
+  readr::read_tsv(file.path(input_data_dir, "ensg-hugo-rmtl-v1-mapping.tsv"),
+                  col_types = readr::cols(.default = readr::col_guess())))
+
+# assert all ensg_ids and gene_symbols are not NA
+if (!identical(sum(is.na(ensg_hugo_rmtl_df$ensg_id)), as.integer(0))) {
+  stop(paste0("Found NA in ensg-hugo-rmtl-v1-mapping.tsv ensg_id.\n",
+              "Check if PedOT release data are downloaded properly.\n",
+              "If data is downloaded properly, submit a GitHub data question."))
+}
+
+if (!identical(sum(is.na(ensg_hugo_rmtl_df$gene_symbol)), as.integer(0))) {
+  stop(paste0("Found NA in ensg-hugo-rmtl-v1-mapping.tsv gene_symbol.\n",
+              "Check if PedOT release data are downloaded properly.\n",
+              "If data is downloaded properly, submit a GitHub data question."))
+}
+
+# assert all ensg_id are unique
+if (!identical(length(unique(ensg_hugo_rmtl_df$ensg_id)),
+               nrow(ensg_hugo_rmtl_df))) {
+  stop(paste0("Found duplicated ensg_id in ensg-hugo-rmtl-v1-mapping.tsv.\n",
+              "Check if PedOT release data are downloaded properly.\n",
+              "If data is downloaded properly, submit a GitHub data question."))
+}
+
+# Download data from https://mygene.info/ --------------------------------------
+message("Retrieve Gene_full_name and Protein_RefSeq_ID from mygene.info...")
+ens_gids <- ensg_hugo_rmtl_df$ensg_id
+
+mg_qres_list <- mygene::queryMany(
+  ens_gids, scopes = "ensembl.gene", fields = c("refseq", "name"),
+  species = "human", returnall = TRUE, return.as = "DataFrame")
+
+found_mg_qres_df <- tibble::as_tibble(
+  mg_qres_list$response[, c("query", "notfound", "name", "refseq.protein")]) %>%
+  tidyr::replace_na(list(notfound = FALSE)) %>%
+  dplyr::filter(!notfound)
+
+# remove rows that have both NA name and NULL refseq.protein
+rm_bnn_found_mg_qres_df <- found_mg_qres_df %>%
+  dplyr::filter(!(is.na(name) & purrr::map_lgl(refseq.protein, is.null)))
+
+# collapses name and refseq.protein for output
+out_rm_bnn_found_mg_qres_df <- rm_bnn_found_mg_qres_df %>%
+  dplyr::group_by(query) %>%
+  dplyr::summarise(name = collapse_name_vec(name),
+                   refseq_protein = collapse_rp_lists(refseq.protein)) %>%
+  dplyr::rename(Gene_Ensembl_ID = query, Gene_full_name = name,
+                Protein_RefSeq_ID = refseq_protein)
+
+
+
+# Output TSV file --------------------------------------------------------------
+readr::write_tsv(
+  out_rm_bnn_found_mg_qres_df,
+  file.path(output_data_dir, "ensg-gene-full-name-refseq-protein.tsv"))
```

Comment:
This file has NAs in the `refseq.protein` column - do we want to keep that here and will you be updating this field upstream, post-annotation to be blanks for FNL JSONL files? I like the idea of keeping NA here and updating to blank for the final files, but just want to be sure this is on the radar.

Reply:
I also prefer keeping NAs in here. The API/CLI provides the `replace_na_with_empty_string`/`--replace-na-with-empty-string` to replace all NAs with `''`.